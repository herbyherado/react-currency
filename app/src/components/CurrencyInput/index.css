/* TOP */
.currency-table {
  border: 1px solid white;
  min-width: 400px;
  max-width: 400px;
}

.currency-input {
  display: flex;
  flex-direction: column;
  padding: 1rem;

  border-bottom: 1px solid white;
}

div.currency-input p {
  margin: 0.5rem 0;
}

p.currency-input-detail {
  font-size: 1rem;
}

div.currency-input-body {
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

div.currency-input-body input {
  font-size: 1.5rem;
  text-align: right;
  color: white;
  background: transparent;
  width: 100%;

  padding: 0.5rem 0.5rem 0.5rem 4rem;
}

div.currency-input-body input[type='number']::-webkit-outer-spin-button,
div.currency-input-body input[type='number']::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

div.currency-input-body input[type='number'] {
  -moz-appearance: textfield;
}

div.currency-input-body p.currency-input-text {
  position: absolute;
  font-weight: 500;
  font-size: 1.5rem;

  display: flex;
  justify-content: center;
  margin-left: 0.5rem;
}
