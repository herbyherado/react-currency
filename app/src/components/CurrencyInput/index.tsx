import React, { PureComponent } from 'react';
import './index.css';

class CurrencyInput extends PureComponent<CurrencyInputProps> {
  handleInputChange = (e: React.FormEvent<HTMLInputElement>) => {
    this.props.onInputChange(e);
  };

  render() {
    const { baseFullName, baseName, baseValue } = this.props;
    return (
      <div className="currency-input">
        <p className="currency-input-detail text-italic">
          {baseName} - {baseFullName}
        </p>
        <div className="currency-input-body">
          <p className="currency-input-text">{baseName}</p>
          <input
            type="text"
            placeholder="Value"
            value={baseValue}
            onChange={this.handleInputChange}
          />
        </div>
      </div>
    );
  }
}

type CurrencyInputProps = {
  baseName: string;
  baseFullName: string;
  baseValue: number | string;
  onInputChange: (event: React.FormEvent<HTMLInputElement>) => void;
};

export default CurrencyInput;
