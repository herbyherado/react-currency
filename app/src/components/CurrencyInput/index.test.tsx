import React from 'react';
import renderer from 'react-test-renderer';
import CurrencyInput from './index';

describe('CurrencyInput component testing', function() {
  describe('CurrencyInput snapshot testing', function() {
    it('should match to snapshot', function() {
      const component: renderer.ReactTestRenderer = renderer.create(
        <CurrencyInput
          baseName="USD"
          baseFullName="United States Dollars"
          baseValue={10}
          onInputChange={jest.fn()}
        />
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe('CurrencyInput method calls', function() {
    describe('handleInputChange', function() {
      it('should call onInputChangeProps', function() {
        const component: renderer.ReactTestRenderer = renderer.create(
          <CurrencyInput
            baseName="USD"
            baseFullName="United States Dollars"
            baseValue={10}
            onInputChange={jest.fn()}
          />
        );

        const mockEvent = { currentTarget: { value: '42' } } as React.FormEvent<HTMLInputElement>;
        const handleInputChange = component.root.instance.handleInputChange;
        const propsSpy: jest.SpyInstance = jest.spyOn(component.root.props, 'onInputChange');

        handleInputChange(mockEvent);
        expect(propsSpy).toHaveBeenCalled();
        expect(propsSpy).toHaveBeenCalledWith(mockEvent);
      });
    });
  });
});
