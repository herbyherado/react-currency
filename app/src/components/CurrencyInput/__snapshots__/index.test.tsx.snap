// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyInput component testing CurrencyInput snapshot testing should match to snapshot 1`] = `
<div
  className="currency-input"
>
  <p
    className="currency-input-detail text-italic"
  >
    USD
     - 
    United States Dollars
  </p>
  <div
    className="currency-input-body"
  >
    <p
      className="currency-input-text"
    >
      USD
    </p>
    <input
      onChange={[Function]}
      placeholder="Value"
      type="text"
      value={10}
    />
  </div>
</div>
`;
